mecanum_drive_controller/test/test_mecanum_drive_controller.cpp

```diff
@@ -0,0 +1,547 @@
+  msg->twist.angular.y = std::numeric_limits<double>::quiet_NaN();
+  msg->twist.angular.z = TEST_ANGULAR_VELOCITY_Z;
+  controller_->input_ref_.writeFromNonRT(msg);
+  const auto age_of_last_command =
+    controller_->get_node()->now() - (*(controller_->input_ref_.readFromNonRT()))->header.stamp;
+
+  // age_of_last_command > ref_timeout_
+  ASSERT_FALSE(age_of_last_command <= controller_->ref_timeout_);
+  ASSERT_EQ((*(controller_->input_ref_.readFromRT()))->twist.linear.x, TEST_LINEAR_VELOCITY_X);
+  ASSERT_EQ(
+    controller_->update_reference_from_subscribers(
+      controller_->get_node()->now(), rclcpp::Duration::from_seconds(0.01)),
+    controller_interface::return_type::OK);
+  ASSERT_EQ(
+    controller_->update_and_write_commands(
+      controller_->get_node()->now(), rclcpp::Duration::from_seconds(0.01)),
+    controller_interface::return_type::OK);
+
+  // EXPECT_TRUE(std::isnan(joint_command_values_[NR_CMD_ITFS - 2]));
+  EXPECT_TRUE(std::isnan(controller_->reference_interfaces_[0]));
+
+  EXPECT_EQ(joint_command_values_[1], 0);
+  // ASSERT_EQ(controller_->reference_interfaces_[0], 0);
+  for (const auto & interface : controller_->reference_interfaces_)
+  {
+    // ASSERT_EQ(interface, 0);
+    EXPECT_TRUE(std::isnan(interface));
+  }
+
+  std::shared_ptr<ControllerReferenceMsg> msg_2 = std::make_shared<ControllerReferenceMsg>();
+  msg_2->header.stamp = controller_->get_node()->now() - rclcpp::Duration::from_seconds(0.01);
+  msg_2->twist.linear.x = TEST_LINEAR_VELOCITY_X;
+  msg_2->twist.linear.y = TEST_LINEAR_VELOCITY_y;
+  msg_2->twist.linear.z = std::numeric_limits<double>::quiet_NaN();
+  msg_2->twist.angular.x = std::numeric_limits<double>::quiet_NaN();
+  msg_2->twist.angular.y = std::numeric_limits<double>::quiet_NaN();
+  msg_2->twist.angular.z = TEST_ANGULAR_VELOCITY_Z;
+  controller_->input_ref_.writeFromNonRT(msg_2);
+  const auto age_of_last_command_2 =
+    controller_->get_node()->now() - (*(controller_->input_ref_.readFromNonRT()))->header.stamp;
+
+  // age_of_last_command_2 < ref_timeout_
+  ASSERT_TRUE(age_of_last_command_2 <= controller_->ref_timeout_);
+  ASSERT_EQ((*(controller_->input_ref_.readFromRT()))->twist.linear.x, TEST_LINEAR_VELOCITY_X);
+  ASSERT_EQ(
+    controller_->update_reference_from_subscribers(
+      controller_->get_node()->now(), rclcpp::Duration::from_seconds(0.01)),
+    controller_interface::return_type::OK);
+  ASSERT_EQ(
+    controller_->update_and_write_commands(
+      controller_->get_node()->now(), rclcpp::Duration::from_seconds(0.01)),
+    controller_interface::return_type::OK);
+
+  EXPECT_NE(joint_command_values_[1], 111);
+  EXPECT_EQ(joint_command_values_[1], 3.0);
+  ASSERT_EQ((*(controller_->input_ref_.readFromRT()))->twist.linear.x, TEST_LINEAR_VELOCITY_X);
+  for (const auto & interface : controller_->reference_interfaces_)
+  {
+    EXPECT_FALSE(std::isnan(interface));
+  }
+}
+
+TEST_F(MecanumDriveControllerTest, test_update_logic)
```

Comment:
Isn't this already tested multiple times? what is new in this test? Please make this part visible

Reply:
there was test_update_logic_chainable and test_update_logic 
only difference between both is a line where it is checked if the controller is in chainable mode or not, in our case it is not.

Reply:
This does not answer the question. What is the purpose of this test compared to the previous two above?

Reply:
Shouldn't here controller be in chainable mode? Since we are testing not chainable mode in the test before?